_posts/2020-09-01-weather-prediction/weather_prediction_deep_learning.Rmd

```diff
@@ -0,0 +1,693 @@
+available at several pressure levels. Thus, our example makes use of a small subset of available "channels". To save storage,
+network and computational resources, it also operates at the smallest available resolution.
+
+This post is accompanied by executable code on [Google
+Colaboratory](https://colab.research.google.com/drive/1URSjWfcnhgesHWCqd7Un5GHECYblY04r?usp=sharing), which should not just
+render unnecessary any copy-pasting of code snippets but also, allow for uncomplicated modification and experimentation.
+
+To read in and extract the data, stored as [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) files, we use
+[tidync](https://docs.ropensci.org/tidync/), a high-level package built on top of
+[ncdf4](https://cran.r-project.org/package=ncdf4) and [RNetCDF](https://cran.r-project.org/package=RNetCDF). Otherwise,
+availability of the usual "TensorFlow family" as well as a subset of tidyverse packages is assumed.
+
+```{r}
+library(reticulate)
+library(tensorflow)
+library(keras)
+library(tfdatasets)
+library(tfautograph)
+
+library(dplyr)
+library(ggplot2)
+library(tidyr)
+library(purrr)
+library(lubridate)
+
+library(tidync)
+```
+
+As already alluded to, our example makes use of two spatio-temporal series: 500hPa geopotential and 850hPa temperature. The
+following commands will download and unpack the respective sets of by-year files, for a spatial resolution of 5.625 degrees:
+
+```{r}
+download.file("https://dataserv.ub.tum.de/s/m1524895/download?path=%2F5.625deg%2Ftemperature_850&files=temperature_850_5.625deg.zip",
+              "temperature_850_5.625deg.zip")
+unzip("temperature_850_5.625deg.zip", exdir = "temperature_850")
+
+download.file("https://dataserv.ub.tum.de/s/m1524895/download?path=%2F5.625deg%2Fgeopotential_500&files=geopotential_500_5.625deg.zip",
+              "geopotential_500_5.625deg.zip")
+unzip("geopotential_500_5.625deg.zip", exdir = "geopotential_500")
+```
+
+Inspecting one of those files' contents, we see that its data array is structured along three dimensions, longitude (64
+different values), latitude (32) and time (8760). The data itself is `z`, the geopotential.
+
+```{r}
+tidync("geopotential_500/geopotential_500hPa_2015_5.625deg.nc") %>% hyper_array()
+```
+
+    Class: tidync_data (list of tidync data arrays)
+    Variables (1): 'z'
+    Dimension (3): lon,lat,time (64, 32, 8760)
+    Source: /[...]/geopotential_500/geopotential_500hPa_2015_5.625deg.nc
+
+Extraction of the data array is as easy as telling `tidync` to read the first in the list of arrays:
+
+```{r}
+z500_2015 <- (tidync("geopotential_500/geopotential_500hPa_2015_5.625deg.nc") %>%
+                hyper_array())[[1]]
+
+dim(z500_2015)
+```
+
+    [1] 64 32 8760
+
+While we delegate further introduction to `tidync` to a comprehensive [blog
+post](https://ropensci.org/blog/2019/11/05/tidync/) on the ROpenSci website, let's at least look at a quick visualization, for
```

Comment:
```suggestion
post](https://ropensci.org/blog/2019/11/05/tidync/) on the ROpenSci website, let's at least look at a quick visualization for
```

Reply:
(removed the comma since it's the visualization for which you're picking the first time point)